wholegames-intro.Rmd

```diff
@@ -0,0 +1,146 @@
+The next section will then present single topics such as "how to deal with authentication" in further details.
+
+## Our example package
+
+Our minimal package, [`exemplighratia`](https://github.com/maelle/exemplighratia), accesses the GitHub status API and one endpoint of GitHub V3 REST API.
+It is named after the Latin phrase _exempli gratia_ that means "for instance", with an H for GH.
+If you really need to interact with GitHub V3 API, we recommend the [gh package](https://gh.r-lib.org/).
+We also recommend looking at the source of the gh package, and at the docs of GitHub V3 API, in particular about [authentication](https://developer.github.com/v3/#authentication).
+
+GitHub V3 API works without authentication too, but at a lower rate.
+For the sake of having an example of a package _requiring_ authentication we shall assume the API is not usable without authentication.
+Authentication is the setting of a token in a HTTP header, so no OAuth dance or so. 
+
+GitHub Status API, on the contrary, does not necessitate authentication at all.
+
+So we have two functions, one that works without authentication, one that works with authentication.
+
+How did we create the package?
+You are obviously free to use your own favorite workflow tools but sharing our workflow might be informative.
+
+* We followed [usethis setup article](https://usethis.r-lib.org/articles/articles/usethis-setup.html).
+
+Then we ran
+
+* `usethis::create_package("path/to/folder/exemplighratia")` to create and open the package project;
+* `usethis::use_mit_license()` to add an MIT license;
+* `usethis::use_package("httr")` to add a dependency on httr;
+* `usethis::use_package("purrr")` to add a dependency on purrr;
+* `use_r("api-status.R")` to add the first function whose code is written below;
+
+```r
+status_url <- function() {
+  "https://kctbh9vrtdwd.statuspage.io/api/v2/components.json"
+}
+
+#' GitHub APIs status
+#'
+#' @description Get the status of requests to GitHub APIs
+#'
+#' @return A character vector, one of "operational", "degraded_performance",
+#' "partial_outage", or "major_outage."
+#'
+#' @details See details in https://www.githubstatus.com/api#components.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' gh_api_status()
+#' }
+gh_api_status <- function() {
+  response <- httr::GET(status_url())
+
+  # Check status
+  httr::stop_for_status(response)
+
+  # Parse the content
+  content <- httr::content(response)
+
+  # Extract the part about the API status
+  components <- content$components
+  api_status <- components[purrr::map_chr(components, "name") == "API Requests"][[1]]
+
+  # Return status
+  api_status$status
+
+}
+```
+
+* `use_test("api-status")` (and using testthat latest version so setting `Config/testthat/edition: 3` in `DESCRIPTION`) to add a simple test whose code is below.
```

Comment:
What is `Config/testthat/edition: 3`? I'm not familiar with that

Reply:
I added a link to https://www.tidyverse.org/blog/2020/10/testthat-3-0-0/#3rd-edition